Source/Lib/Codec/EbEncHandle.c

```diff
@@ -3365,27 +3375,37 @@ static EB_ERRORTYPE CopyFrameBuffer(
 
 
         // Y
-        for (inputRowIndex = 0; inputRowIndex < lumaHeight; inputRowIndex++) {
-
-            EB_MEMCPY((inputPicturePtr->bufferY + lumaBufferOffset + lumaStride * inputRowIndex),
-                (inputPtr->luma + sourceLumaStride * inputRowIndex),
-                lumaWidth);
+        if (sequenceControlSetPtr->lcuAligned)
+            inputPicturePtr->bufferY = inputPtr->luma;
```

Comment:
this assumes that the layout of the input buffer is what the library is expecting, such might be the case with the svt sample app but not other applications as decoder buffers might use a different padding structure.

Reply:
Yes, because the ffmpeg libsvt_hevc plugin (based on pure software encoder) doesn't implemente the callbacks of HWContextType, like what other hardware acceleration encoders do, ffmpeg doesn't know the buffer requirement (such as stride alignment) of SVT-HEVC encoder. And need to check whether GStreamer has the similar behavior as well.